Src/Plugins/Security/SEC101_050.IdentifiableNpmAuthorTokenValidator.cs

```diff
@@ -0,0 +1,62 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.Net.Http;
+using System.Text;
+
+using Microsoft.CodeAnalysis.Sarif.PatternMatcher.Sdk;
+using Microsoft.RE2.Managed;
+using Microsoft.Security.Utilities;
+
+namespace Microsoft.CodeAnalysis.Sarif.PatternMatcher.Plugins.Security
+{
+    public class IdentifiableNpmAuthorTokenValidator : DynamicValidatorBase
+    {
+        [ThreadStatic]
+        private static StringBuilder s_sb;
+
+        protected override IEnumerable<ValidationResult> IsValidStaticHelper(IDictionary<string, FlexMatch> groups)
+        {
+            FlexMatch secret = groups["secret"];
+            FlexMatch checksum = groups["checksum"];
+
+            // Validate checksum to avoid false positives.
+            string randomPart = secret.Value.String.Substring(4, 30);
+            uint checksumValue = Crc32.Calculate(randomPart);
+            var encoder = new CustomAlphabetEncoder();
```

Comment:
>var encoder = new CustomAlphabetEncoder(); [](http://example.com/codeflow?start=12&length=42)

should we create only one encoder and re-use it?

Reply:
Thanks, this also helped me catch some other cleanup items.